spec/requests/api/v2/exchange_rates/files_controller_spec.rb

```diff
@@ -79,15 +81,28 @@
       end
     end
 
-    context 'when requesting invalid type' do
-      let(:type) { 'non_existing_type' }
+    context 'when file not found' do
+      let(:period_type) { 'monthly_csv' }
       let(:format) { 'csv' }
 
-      before { get api_exchange_rates_file_path("#{type}_#{year}-#{month}", format: :csv) }
+      let(:wrong_year) { 2010 }
 
-      it 'returns HTTP status :not_found' do
+      before { get api_exchange_rates_file_path("#{period_type}_#{wrong_year}-#{month}", format: :csv) }
+
+      it 'returns HTTP 404  :not_found' do
         expect(response).to have_http_status(:not_found)
       end
     end
+
+    context 'when requesting invalid period type' do
+      let(:period_type) { 'invalid_period_type' }
+      let(:format) { 'csv' }
+
+      before { get api_exchange_rates_file_path("#{period_type}_#{year}-#{month}", format:) }
+
+      it 'returns HTTP 400 bad request' do
+        expect(response).to have_http_status(:bad_request)
+      end
```

Comment:
This is the new behaviour that has been added.
When the period_type is wrong, it returns bad request 400.